tests/test_experiment.py

```diff
@@ -1251,20 +1259,53 @@ def test_experiment_27_run_once_with_optimizer(self):
         self.test_exp.set_agg_optimizer(optim)
         # Disable breakpointing.
         self.test_exp.set_save_breakpoints(False)
-        # Call `Experiment.run_once`, hijacking declearn Vectors.
-        with patch("declearn.model.api.Vector.build") as patch_vector_build:
-            mock_vector = create_autospec(Vector, instance=True)
-            patch_vector_build.return_value = mock_vector
-            self.test_exp.run_once()
+        # Call `Experiment.run_once`, hijacking declearn Vectors
+        # due to the use of empty dicts of (mock) weights.
+        with patch.object(Vector, "build", new=create_autospec(Vector)):
+            result = self.test_exp.run_once()
         # Verify that expected operations occured.
-        self.assertEqual(patch_vector_build.call_count, 2)
-        optim.init_round.assert_called_once()
-        optim.step.assert_called_once()
+        self.assertEqual(result, 1, "run_once did not successfully run the round")
+        self.assertListEqual(
+            [name for name, *_ in optim.method_calls],
+            ["get_aux", "set_aux", "init_round", "step"],
+            "aggregates Optimizer did not receive expected ordered calls"
+        )
+
+    def test_experiment_28_run_once_fails_from_unexpected_nodes_aux_var(self):
```

Comment:
I guess we are missing a test that tests run_once with an optimizer that received `aux_var` and another that check that loading optimizer from breakpoint with `aux_var` gives the correct `aux_var`

Reply:
> I guess we are missing a test that tests run_once with an optimizer that received aux_var (could be appended to already existing test_experiment_27)

This is already tested in `test_experiment_27_run_once_with_optimizer`, using an autospec mock `Optimizer`: we test that the methods corresponding to collecting/sending, and later unpacking/processing auxiliary variables are each called once, and in the expected order.

> and another that check that loading optimizer from breakpoint with aux_var gives the correct aux_var

We already test that loading and Optimizer restores its internal states. As aux_var are emitted deterministically based on these states, there is no more thing to test.

To me, there are no missing things in the generic/general unit tests. There would however be value in setting up an integration test using Optimizer-based Scaffold - which would be a way to conduct these verification in practice.

Reply:
> We already test that loading and Optimizer restores its internal states. As aux_var are emitted deterministically based on these states, there is no more thing to test.

But we don't make sure that Optimizer saved through a breakpoint file is loaded correctly, since it involves a few files (mainly Optimizer content file, and breakpoint state): my feeling would be to test that in a "integration test"

Reply:
added